src/homeworkTenth/MyHashMap.java

```diff
@@ -0,0 +1,81 @@
+package homeworkTenth;
+
+public class MyHashMap<K, V> {
+
+    private Entry<K, V>[] buckets;
+    private static final int INITIAL_CAPACITY = 1 << 4;
+    private int size = 0;
+
+    public MyHashMap() {
+        this(INITIAL_CAPACITY);
+    }
+
+    public MyHashMap(int capacity) {
+        this.buckets = new Entry[capacity];
+    }
+
+    public void put(K key, V value) {
+        Entry<K, V> entry = new Entry<>(key, value, null);
+        int bucket = getHash(key) % getBucketSize();
+
+        Entry<K, V> exiting = buckets[bucket];
+        if (exiting == null) {
+            buckets[bucket] = entry;
+            size++;
+        } else {
+            while (exiting.next != null) {
+                if (exiting.key.equals(key)) {
+                    exiting.value = value;
+                    return;
+                }
+                exiting = exiting.next;
+            }
+            if (exiting.key.equals(key)) {
+                exiting.value = value;
+            } else {
+                exiting.next = entry;
+                size++;
+            }
+        }
+    }
+
+    public V get(K key) {
+        Entry<K, V> bucket = buckets[getHash(key) % getBucketSize()];
+
+        while (bucket != null) {
+            if (key == bucket.key) {
+                return bucket.value;
+            }
+            bucket = bucket.next;
+        }
+        return null;
```

Comment:
??

Reply:
we are computing the hash code from the key, and then compute the index from the hash code with module operation. 
Then, get the linked list at index computed above and search through the linked list for the value with this value.